Makefile.am

```diff
@@ -109,7 +110,7 @@ tools_tpm2_clearlock_SOURCES = tools/tpm2_clearlock.c $(TOOL_SRC)
 tools_tpm2_create_SOURCES = tools/tpm2_create.c $(TOOL_SRC)
 tools_tpm2_createprimary_SOURCES = tools/tpm2_createprimary.c $(TOOL_SRC)
 tools_tpm2_getcap_SOURCES = tools/tpm2_getcap.c $(TOOL_SRC)
-tools_tpm2_pcrlist_SOURCES = tools/tpm2_pcrlist.c $(TOOL_SRC)
+tools_tpm2_pcrlist_SOURCES = tools/tpm2_pcrlist.c $(ETOOL_SRC)
```

Comment:
It would be a little more work, but could we just create a union with both contexts and send them to the tools?
Or maybe have a tool say, give me esapi and send it the esapi vs sapi context?

I don't want to have near duplicate files hanging around.

Reply:
you mean something like this and a bunch of other ifdefs thoughout the tpm2-tools file ?
```
#ifdef SAPI
#define TOOL_CONTEXT TSS2_SYS_CONTEXT
#elif defined ESAPI
#define TOOL_CONTEXT ESYS_CONTEXT
#else
#error "Need to specify -DSAPI or -DESAPI !"
#endif
```

Reply:
No, pass this to the tools "on run" routine.
```c
union {
 ESAPI *esapi;
 SAPI *sapi;
} tpm_context;
```
You'll have to update tpm2_tool.c to instantiate both of these and update the interface to all tools to pass this. Then in the tools just assign the original variable name from the union.

If you don't want to instantiate both, you can just have tpm2_options_new() set a flag to indicate to give me esapi, and then pick based on that.

Reply:
I don't think I understand you. Do you wish to keep both implementations (SAPI and ESAPI) side-by-side, or are you requesting less duplicate code for the transition period from SAPI to ESAPI ?

In the latter case, I'd prefer to have each of the ESAPI files look like they shall look in the end, so we don't need to touch them again once SAPI is eliminated; i.e. they all use ESYS_CONTEXT.
The tpm2_tools.c can would be adapted to SAPI vs ESAPI using #ifdefs that we then remove after transition is complete ?